specification/programmableconnectivity/Azure.ProgrammableConnectivity/apis/location.tsp

```diff
@@ -0,0 +1,73 @@
+import "./common.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+
+namespace Azure.ProgrammableConnectivity;
+
+/// Interfaces
+
+interface DeviceLocation {
+  @doc("Verifies whether a device is within a specified location area, defined as an accuracy (radius) around a point, specified by longitude and latitude.")
+  verify is Operations.ResourceAction<
+    DeviceLocationEndpoint,
+    DeviceLocationVerificationContent,
+    DeviceLocationVerificationResult
+  >;
+}
+
+/// Endpoints
+
+@resource("device-location")
+@doc("Static endpoint to access the Device Location API family.")
+model DeviceLocationEndpoint {
+  @key
+  @doc("Static endpoint")
+  @visibility("read")
+  location: "location";
+}
+
+/// Request models
+
+@doc("Request to verify Location")
+model DeviceLocationVerificationContent {
+  @doc("Network to query for this device, or device information to enable network routing.")
+  networkIdentifier: NetworkIdentifier;
+
+  @doc("Latitude of location to be verified")
+  @minValue(-90)
+  @maxValue(90)
+  latitude: float64;
+
+  @doc("Longitude of location to be verified")
+  @minValue(-180)
+  @maxValue(180)
+  longitude: float64;
+
+  @doc("Accuracy expected for location verification in kilometers")
+  @minValue(2)
+  @maxValue(100)
+  accuracy: int32;
+
+  @doc("The device to find the location for.")
+  locationDevice: LocationDevice;
+}
+
+/// Response models
+
+@doc("Response verifying location")
+model DeviceLocationVerificationResult {
+  @doc("True if the location is in the specified area, False otherwise")
+  verificationResult: boolean;
+}
+
+/// Common models
+@doc("Device information needed by operator to provide location information. Include exactly one of these properties to identify your device.")
+model LocationDevice {
+  ...NetworkAccessIdentifierModel;
```

Comment:
Everything inside the spread models is optional - if they are mutually exclusive this could this be modeled as a union

Reply:
Unfortunately this gives the warning "Unions cannot be emitted to OpenAPI v2 unless all options are literals of the same type". I tried both standard unnamed union syntax (i.e. `alias LocationDevice = NetworkAccessIdentifierModel | PhoneNumberModel | ...`) and also using a named union decorated with @discriminator, and both of them gave me that same error. 

In an [earlier comment](https://microsoft.sharepoint.com/:w:/r/teams/CommunicationsSDK/Shared%20Documents/Alpaca%20-%20(Gateway)%20-%20General/API%20and%20SDK%20Review%20Processes/APC%20API%20questions%20.docx?d=we5538ba2edea4887a78ab582719792e9&csf=1&web=1&e=FlcGIP&nav=eyJjIjoxOTIxMzQ2Mzk1fQ%3D%3D) you suggested that it's simpler to just document the discriminated union - I've added a bit more clarity to the docstring on the locationDevice field

Reply:
Hm, the SPO link doesn't work for me. But I guess I had linked https://typespec.io/docs/standard-library/discriminated-types#using-polymorphism? The discriminator polymorphic style should work technically. We can merge it like it is now and revisit later for reevaluation in a new version.

Reply:
Yes, that's one of the syntaxes I tried that gave me the error above. We could suppress it, but yes, I agree better to just merge now and revisit later